packages-ts/gauntlet-terra-contracts/src/commands/abstract/batchWrapper.ts

```diff
@@ -0,0 +1,62 @@
+import AbstractCommand, { makeAbstractCommand } from '.'
+import { Result, WriteCommand } from '@chainlink/gauntlet-core'
+import { TerraCommand, TransactionResponse, logger } from '@chainlink/gauntlet-terra'
+import { AccAddress, LCDClient, Msg } from '@terra-money/terra.js'
+import { prompt } from '@chainlink/gauntlet-core/dist/utils'
+import { ExecutionContext } from './executionWrapper'
+import { parseJSON } from '@chainlink/gauntlet-terra/dist/lib/rdd'
+
+export const wrapCommand = (command) => {
+  return class BatchCommand extends TerraCommand {
+    static id = `${command.id}:batch`
+    subCommands: any[]
+
+    constructor(flags, args) {
+      super(flags, args)
+    }
+
+    buildCommand = async (flags, args): Promise<TerraCommand> => {
+      const input = flags.input ? flags.input : parseJSON(flags.inputFile, 'BatchInputFile')
+
+      this.subCommands = await Promise.all(
+        input.map(async (individualInput) => {
+          const newFlags = { ...flags, input: individualInput }
+
+          let c = new command(newFlags, args) as TerraCommand
+          await c.invokeMiddlewares(c, c.middlewares)
+          c = c.buildCommand ? await c.buildCommand(newFlags, args) : c
+          return c
+        }),
+      )
+      return this
+    }
+
+    makeRawTransaction = async (signer: AccAddress) => {
+      return await Promise.all(this.subCommands.map(async (element) => (await element.makeRawTransaction(signer))[0]))
+    }
+
+    execute = async (): Promise<Result<TransactionResponse>> => {
+      // TODO: Command should be built from gauntet-core
+      await this.buildCommand(this.flags, this.args)
+      await Promise.all(this.subCommands.map(async (element) => await element.command.simulateExecute()))
+
+      logger.loading(`Executing ${command.id} from contract ${this.args[0]} for the following sets of inputs`)
+      let x = this.subCommands.map((element) => element.command.params)
+      logger.log('Input Params:', x)
+      await prompt(`Continue?`)
+
+      const msgs = await this.makeRawTransaction(this.subCommands[0].wallet.key.accAddress)
+      let tx = await this.subCommands[0].signAndSend(msgs)
```

Comment:
This is still a Terra Command, we should have these fns available
```suggestion
      const msgs = await this.makeRawTransaction(this.wallet.key.accAddress)
      const tx = await this.signAndSend(msgs)
```

Reply:
These functions aren't available until we call `invokeMiddlewares` on the command

Reply:
`invokeMiddlewares` is executed by `gauntlet-core` when building the command. They should be available